stacker/actions/diff.py

```diff
@@ -0,0 +1,181 @@
+import logging
+
+from .. import exceptions
+from ..plan import COMPLETE, SKIPPED, Plan
+from . import build
+import difflib
+import json
+
+logger = logging.getLogger(__name__)
+
+
+def diff_dictionaries(old_dict, new_dict):
+    """Diffs two single dimension dictionaries
+
+    Returns the number of changes and an unordered list
+    expressing the common entries and changes.
+
+    Args:
+        old_dict(dict): old dictionary
+        new_dict(dict): new dictionary
+
+    Returns: list()
+        int: number of changed records
+        list: [str(<change type>), <key>, <value>]
+
+        Where <change type>: +, - or empty string
+    """
+
+    old_set = set(old_dict)
+    new_set = set(new_dict)
+
+    added_set = new_set - old_set
+    removed_set = old_set - new_set
+    common_set = old_set & new_set
+
+    changes = 0
+    output = []
+    for key in added_set:
+        changes += 1
+        output.append(['+', key, new_dict[key]])
+
+    for key in removed_set:
+        changes += 1
+        output.append(['-', key, old_dict[key]])
+
+    for key in common_set:
+        if str(old_dict[key]) != str(new_dict[key]):
+            changes += 1
+            output.append(['-', key, old_dict[key]])
+            output.append(['+', key, new_dict[key]])
+        else:
+            output.append(['', key, new_dict[key]])
+
+    return [changes, output]
+
+
+class Action(build.Action):
+    """ Responsible for diff'ing CF stacks in AWS and on disk
+
+    Generates the build plan based on stack dependencies (these dependencies
+    are determined automatically based on references to output values from
+    other stacks).
+
+    The plan is then used to pull the current CloudFormation template from
+    AWS and compare it to the generated templated based on the current
+    config.
+    """
+
+    def _diff_parameters(self, old_params, new_params):
+        """Compares the old vs. new parameters and prints a 'diff'
+
+        If there are no changes, we print nothing.
+
+        Args:
+            old_params(dict): old paramters
+            new_params(dict): new parameters
+        """
+        [changes, diff] = diff_dictionaries(old_params, new_params)
+        if changes == 0:
+            return
+
+        print """*** New Parameters
```

Comment:
I used the exact same header format that difflib generates for it's diff for the template.  Yes, they're different diff types, but this makes it easier to parse if for some reason someone wanted to do that.